synthesizer/src/vm/deploy.rs

```
@@ -32,9 +32,15 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
         rng: &mut R,
     ) -> Result<Transaction<N>> {
         // Compute the deployment.
-        let deployment = self.deploy_raw(program, rng)?;
+        let mut deployment = self.deploy_raw(program, rng)?;
         // Ensure the transaction is not empty.
         ensure!(!deployment.program().functions().is_empty(), "Attempted to create an empty transaction deployment");
+        // Unset the checksum if the `CONSENSUS_VERSION` is less than `V5`.
```

Hmm, if the program contains a constructor, it will still be rejected by the recipient node.

What about we "fail loudly" and return an error here, if the query is below V5 and the program contains a constructor?

---

The checksum must also be unset for programs without a constructor.